src/diffusers/models/modeling_utils.py

```diff
@@ -729,8 +800,19 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                 with accelerate.init_empty_weights():
                     model = cls.from_config(config, **unused_kwargs)
 
+                if hf_quantizer is not None:
+                    hf_quantizer.preprocess_model(
+                        model=model, device_map=device_map, keep_in_fp32_modules=keep_in_fp32_modules
+                    )
+
+                    # We store the original dtype for quantized models as we cannot easily retrieve it
+                    # once the weights have been quantized
+                    # Note that once you have loaded a quantized model, you can't change its dtype so this will
+                    # remain a single source of truth
+                    config["_pre_quantization_dtype"] = torch_dtype
+
                 # if device_map is None, load the state dict and move the params from meta device to the cpu
-                if device_map is None and not is_sharded:
+                if device_map is None and not is_sharded or (hf_quantizer is not None):
```

Comment:
@SunMarc I don't have a better check at the moment to make it work. Without this, the code hits the other path with `dispatch_and_load_checkpoint()` complaining about (which I guess is expected):

```bash
ValueError: Blockwise quantization only supports 16/32-bit floats, but got torch.uint8
```

We could handle this case separately i.e., when the quantized checkpoints are sharded and unsupported by `accelerate` at the moment?